cgi-bin/DW/Controller/Inbox.pm

```diff
@@ -0,0 +1,659 @@
+package DW::Controller::Inbox;
+use strict;
+use warnings;
+
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+use DW::FormErrors;
+use LJ::Hooks;
+use Data::Dumper;
+
+DW::Routing->register_string( '/inbox',          \&index_handler,    app => 1 );
+DW::Routing->register_string( '/inbox/compose',  \&compose_handler,  app => 1 );
+DW::Routing->register_string( '/inbox/markspam', \&markspam_handler, app => 1 );
+DW::Routing->register_rpc( "inbox_actions", \&action_handler, format => 'html' );
+
+my $PAGE_LIMIT = 15;
+
+sub index_handler {
+    my ( $ok, $rv ) = controller( form_auth => 1 );
+    return $rv unless $ok;
+
+    my $r      = $rv->{r};
+    my $POST   = $r->post_args;
+    my $GET    = $r->get_args;
+    my $remote = $rv->{remote};
+    my $vars;
+    my $scope = '/inbox/index.tt';
+
+    return error_ml("$scope.error.not_ready") unless $remote->can_use_esn;
+
+    my $inbox = $remote->notification_inbox
+        or return error_ml( "$scope.error.couldnt_retrieve_inbox", { 'user' => $remote->{user} } );
```

Comment:
nit: Quoting `'user'` here is unnecessary, and we prefer not to stylistically (reduces clutter). You only need to quote keys in hashes when the keys start with numbers or contain certain symbols.

Reply:
I just realized that a lot of this code came from the BML. You can ignore most of the nits... I don't need us to clean up the Perl we're copying from BML...